File: app/models/qdm/structure_map.rb
module QDM
  # app/models/qdm/structure_map.rb
  class StructureMap < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :url
    field :identifier
    field :version
    field :name
    field :title
    field :status
    field :experimental
    field :date
    field :publisher
    field :contact
    field :description
    field :useContext
    field :jurisdiction
    field :purpose
    field :copyright
    field :structure
    field :import
    field :group
    field :qdmVersion, type: String, default: '4.0.0'
  end
end

File: app/assets/javascripts/StructureMap.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const StructureMapSchema = DataElementSchema({
  url: Any,
  identifier: Any,
  version: Any,
  name: Any,
  title: Any,
  status: Any,
  experimental: Any,
  date: Any,
  publisher: Any,
  contact: Any,
  description: Any,
  useContext: Any,
  jurisdiction: Any,
  purpose: Any,
  copyright: Any,
  structure: Any,
  import: Any,
  group: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'StructureMap' },

});

module.exports.StructureMapSchema = StructureMapSchema;
module.exports.StructureMap = mongoose.model('StructureMap', StructureMapSchema);
